ci/docker-entrypoint.sh

```diff
@@ -13,4 +13,14 @@ if [ "$1" = 'envoy' ]; then
 	fi
 fi
 
-exec "$@"
+if [ "$ENVOY_UID" != "0" ]; then
```

Comment:
What is the default here? If not set somehow by default should this default to "0"? I think by default we will enter this statement but the string will be empty?

Reply:
thats correct. The default is to use the `envoy` user with the `uid`/`gid` that is set at build time

Reply:
they are runtime env vars for the container - so they can be set eg on the `docker` command line or in a `compose` file etc

Reply:
It's not clear that this now has to be set. Can we please change this so that the default stays wat it was and we document how to set the right variables? By default I don't think this makes sense.

Reply:
If both of these env vars are empty strings which would be the defualt, do the commands even work?

Reply:
yep for sure. 

the default case is that `ENVOY_UID` is not set - ie also not set to `0`

in this case it goes into this code block, but as `ENVOY_UID` is not set (same with gid) it goes straight to
```
     su-exec envoy "${@}"
```

Reply:
Ah OK, gotcha. Alright thanks for the explanation. LGTM.